File: Non_Repeating_Character.java
class Solution
{
    //Function to find the first non-repeating character in a string.
    static char nonrepeatingCharacter(String S)
    {
        //Your code here
        HashMap<Character,Integer> h=new HashMap<Character,Integer>();
        for( int i=0;i< S.length();i++)
        {
            char x=S.charAt(i);
            h.put(x, h.getOrDefault(x,0)+1);
        }
        for( int i=0;i< S.length();i++)
        {
            char x=S.charAt(i);
            if (h.get(x)==1)
            {
                return x;
            }
        }
        return '$';
    }
}

File: Difficulty: Easy/Non Repeating Character/non-repeating-character.py
#User function Template for python3
from collections import Counter
class Solution:
    
    #Function to find the first non-repeating character in a string.
    def nonRepeatingChar(self,s):
        #code here
        c=Counter(s)
        for x in s :
            if c[x]==1 :
                return x
        return -1
    


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER


@atexit.register
def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())


if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        s = str(input())
        obj = Solution()
        ans = obj.nonRepeatingChar(s)
        if (ans != '$'):
            print(ans)
        else:
            print(-1)

        print("~")

# } Driver Code Ends